docs/proposals/20231031-rdt-enhanced.md

```diff
@@ -0,0 +1,174 @@
+any additional information provided beyond the standard proposal template.
+[Tools for generating](https://github.com/ekalinin/github-markdown-toc) a table of contents from markdown are available.
+
+- [Title](#title)
+  - [Table of Contents](#table-of-contents)
+  - [Glossary](#glossary)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals/Future Work](#non-goalsfuture-work)
+  - [Proposal](#proposal)
+    - [User Stories](#user-stories)
+      - [Story 1](#story-1)
+      - [Story 2](#story-2)
+    - [Requirements (Optional)](#requirements-optional)
+      - [Functional Requirements](#functional-requirements)
+        - [FR1](#fr1)
+        - [FR2](#fr2)
+      - [Non-Functional Requirements](#non-functional-requirements)
+        - [NFR1](#nfr1)
+        - [NFR2](#nfr2)
+    - [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+    - [Risks and Mitigations](#risks-and-mitigations)
+  - [Alternatives](#alternatives)
+  - [Upgrade Strategy](#upgrade-strategy)
+  - [Additional Details](#additional-details)
+    - [Test Plan [optional]](#test-plan-optional)
+  - [Implementation History](#implementation-history)
+
+## Glossary
+
+RDT, Resource Directory Technology. See: https://www.intel.com/content/www/us/en/architecture-and-technology/resource-director-technology.html
+
+## Summary
+
+We hope to enhance RDT usage in koordinator to import pod level RDT adjustment in realtime and integrate RDT monitor feature into koordinator.
+
+## Motivation
+
+Koordinator support RDT configuration and adjustment by config map based on class level. It uses a goroutine to set/adjust RDT configuration in async mode which may not in real time. As koordinator already support NRI in 0.3.0 release, we can migrate current function into koordlet as an runtime hook plugin which may be more real time. Also we want to enhance RDT feature in the same time which include pod level RDT configure/adjustment and monitor. With these features, koordinator can have more information to determine how to adjust different workloads' RDT or even other resources' configuration.
+
+### Goals
+
+- Migrate existed RDT function into runtime hook based on NRI
+- Add pod level RDT configuration/adjustment
+- Add monitor for class and pod level
+
+### Non-Goals/Future Work
+
+- RDT configure policy
+- Schedule workload based on RDT resource
+- AMD ResCtl support
+
+## Proposal
+
+We will implement RDT related function as a runtime hook plugin. RDT runtime hook plugin will still watch a configmap and generate UpdateContainer CRI request to dynamically adjust RDT configuration. In the meanwhile, RDT runtime hook plugin will create RDT group and monitor group first for each class during Pod RunPodSandBox and modify OCI spec for pod and container. We rely on runc to make RDT configuration work. A. We will collect RDT resource metrics and report them to koordlet.
```

Comment:
Dynamic create group will make the NRI-plugin advantage less obvious?

Reply:
I mean we still use NRI to dymically create group when first pod in this group is created.

Reply:
> I mean we still use NRI to dymically create group when first pod in this group is created.

We can talk about more details in the afternoon. 😀